src/workers/dataFetcher.ts

```diff
@@ -22,146 +22,167 @@ class JobData extends Job {
 
 const queryStruct = z.object({
   sql: z.string(),
-  paramsToReplace: z.array(
-    z.object({
-      param: z.string(),
-      placeholder: z.string(),
-    })
-  ),
+  // paramsToReplace: z.array(
+  //   z.object({
+  //     param: z.string(),
+  //     placeholder: z.string(),
+  //   })
+  // ),
 })
 
 const worker = new Worker(
   'dataFetcher',
   async (job: JobData) => {
     console.log(`DATAFETCHER JOB, Attempt #${job.attemptsMade}`, job.data)
-    job.log(`DATAFETCHER JOB, Attempt #${job.attemptsMade}` + job.data)
-    const {
-      threadChannelId,
-      distilledQuestion,
-      plan,
-      strAnlNr,
-      msgId,
-      brokenSql,
-      sqlError,
-    } = job.data
+    console.log(`DATAFETCHER JOB, Attempt #${job.attemptsMade}` + job.data)
+    const { threadChannelId, distilledQuestion, plan, strAnlNr, msgId } =
+      job.data
     let typingHandle: NodeJS.Timeout
     try {
       const thread = (await discord.channel(threadChannelId)) as TextChannel
       const msg = await thread.messages.fetch(msgId)
-      await msg.edit('Skapar databasfrågor')
-      job.log('Skapar databasfrågor')
+      console.log('Skapar databasfrågor')
 
       typingHandle = setInterval(() => {
         thread.sendTyping()
       }, 8000)
 
       const json_mode = true
-      const systemPrompt = sqlPrompt(brokenSql, sqlError)
-      const userPrompt = distilledQuestion + plan
-      job.log('System prompt: ' + systemPrompt)
-      job.log('User prompt: ' + userPrompt)
-      const sqlQueryResponse = await createCompletion(
-        [
-          {
-            role: 'system',
-            content: systemPrompt,
-          },
-          {
-            role: 'user',
-            content: userPrompt,
-          },
-        ],
-        json_mode
-      )
-      clearInterval(typingHandle)
+      let brokenSql = ''
+      let sqlError = ''
+      let userPrompt = distilledQuestion
+      let sql = undefined
+      let json = undefined
+      let results = []
+      let parsed: z.SafeParseReturnType<any, any> | undefined = undefined
+      let queryGenerationAttempts = 0
+      const attemptLimit = 10
+      while (!json || !parsed?.success) {
```

Comment:
Give total ten attempts on wherever it fails, JSON gen, SQL gen, execution... Feedback errors in while loop incrementally.